client/apps/settings/view-wrapper.js

```diff
@@ -116,6 +118,7 @@ const saveChanges = ( successAction, failureAction ) => ( dispatch, getState ) =
 	// Trigger a full validation
 	dispatch( closeShippingZoneMethod( siteId ) );
 	if ( getCurrentlyOpenShippingZoneMethod( getState() ) ) {
+		dispatch( failureAction );
```

Comment:
It's not apparent to me why we need to dispatch our failure action here.

Also, I know this wasn't introduced in your changes, but do you know why we aren't using the `method` variable in the `if`?

Reply:
I've added some comments to clarify that behaviour. Basically, in this function we are "emulating" the Calypso UI and behaviour.

When dispatching the `closeShippingMethod()` action, it triggers a full validation of the shipping method, because that's what happens when you close the shipping method modal in Calypso. The Redux state tree knows which method is currently opened so there's no need to specify `method.id`.

If the validation fails, the shipping method isn't closed (in Calypso that would mean that the modal would keep being opened), so if after dispatching the `close` action there is still a method marked as being edited, we know it failed.